packages/wrangler/src/assets.ts

```diff
@@ -478,75 +531,44 @@ export function validateAssetsArgsAndConfig(
 				"Read more: https://developers.cloudflare.com/workers/static-assets/binding/#smart-placement"
 		);
 	}
-
-	// Provided both the run_worker_first and experimental_serve_directly options
-	if (
-		"legacy" in args
-			? args.assets?.assetConfig?.run_worker_first !== undefined &&
-				args.assets?.assetConfig.serve_directly !== undefined
-			: config?.assets?.run_worker_first !== undefined &&
-				config?.assets?.experimental_serve_directly !== undefined
-	) {
-		throw new UserError(
-			"run_worker_first and experimental_serve_directly specified.\n" +
-				"Only one of these configuration options may be provided."
-		);
-	}
-
-	// User Worker ahead of assets, but no assets binding provided
-	if (
-		"legacy" in args
-			? args.assets?.assetConfig?.run_worker_first === true &&
-				!args.assets?.binding
-			: config?.assets?.run_worker_first === true && !config?.assets?.binding
-	) {
-		logger.warn(
-			"run_worker_first=true set without an assets binding\n" +
-				"Setting run_worker_first to true will always invoke your Worker script.\n" +
-				"To fetch your assets from your Worker, please set the [assets.binding] key in your configuration file.\n\n" +
-				"Read more: https://developers.cloudflare.com/workers/static-assets/binding/#binding"
-		);
-	}
-
-	// Using run_worker_first=true, but didn't provide a Worker script
-	if (
-		"legacy" in args
-			? args.entrypoint === noOpEntrypoint &&
-				args.assets?.assetConfig?.run_worker_first === true
-			: !config?.main && config?.assets?.run_worker_first === true
-	) {
-		throw new UserError(
-			"Cannot set run_worker_first=true without a Worker script.\n" +
-				"Please remove run_worker_first from your configuration file, or provide a Worker script in your configuration file (`main`)."
-		);
-	}
 }
 
 const CF_ASSETS_IGNORE_FILENAME = ".assetsignore";
+const REDIRECTS_FILENAME = "_redirects";
+const HEADERS_FILENAME = "_headers";
 
 /**
  * Create a function for filtering out ignored assets.
  *
  * The generated function takes an asset path, relative to the asset directory,
  * and returns true if the asset should not be ignored.
  */
-async function createAssetIgnoreFunction(dir: string) {
+export async function createAssetsIgnoreFunction(dir: string) {
 	const cfAssetIgnorePath = path.resolve(dir, CF_ASSETS_IGNORE_FILENAME);
 
-	if (!existsSync(cfAssetIgnorePath)) {
-		return null;
+	const ignorePatterns = [
+		// Ignore the `.assetsignore` file and other metafiles by default.
+		// The ignore lib expects unix-style paths for its patterns
```

Comment:
Bashed my head against the wall for quite a while here to get the Windows tests passing. Good thing we have them!